//
// Copyright 2019 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo;

import "asylo/enclave.proto";

// Input to a fork security test, specifying whether to request a snapshot
// inside the enclave.
message ForkSecurityTestInput {
  // Whether request a snapshot inside the enclave.
  optional bool request_fork = 1;
  enum ThreadType {
    UNSUPPORTED = 0;
    // Enters the enclave to set |request_fork| bit.
    SETREQUEST = 1;
    // Enters the enclave and busy wait.
    WAIT = 2;
  }
  optional ThreadType thread_type = 2;

  // An untrusted address that specifies whether the wait thread is staying
  // inside the enclave.
  optional uint64 wait_thread_inside = 3;
}

extend EnclaveInput {
  optional ForkSecurityTestInput fork_security_test_input = 243359234;
}
